### PR TITLE
Adding update graph documentation

### PR DESCRIPTION
**Pull request by gallettilance**

### CODE REVIEW

File: content/en/docs/Concepts/olm-architecture/operator-registry/building-a-catalog.md [hunk 1]
```diff
@@ -77,6 +77,105 @@ opm index add --bundles quay.io/my-container-registry-namespace/my-manifest-bund
...
+OLM also allows you to specify updates through version ranges in your CSV.
+
+```yaml
+metadata:
+   name: myoperator.v1.0.3
+   annotations:
+      olm.skiprange: <1.0.3
+```
+
+The version specifying the `skipRange` will be presented as a direct (one hop) update to
```
Comment (ankitathomas) [inline content/en/docs/Concepts/olm-architecture/operator-registry/building-a-catalog.md:146]: nit: This should be olm.skipRange

File: content/en/docs/Concepts/olm-architecture/operator-registry/building-a-catalog.md [hunk 2]
```diff
@@ -77,6 +77,106 @@ opm index add --bundles quay.io/my-container-registry-namespace/my-manifest-bund
...
+the chain. For example, given
+
+```txt
+myoperator.v1.0.0 -> myoperator.v1.0.1 -> myoperator.v1.0.2
+```
+
+A subscription on `myoperator.v1.0.0` will update to `myoperator.v1.0.2` through `myoperator.v1.0.1`.
+
+Installing from the UI today will always install the latest of a given channel. However, installing
+specific versions is possible with this update graph by modifying the `startingCsv` field
```
Comment (dinhxuanvu) [inline content/en/docs/Concepts/olm-architecture/operator-registry/building-a-catalog.md:108]: `startingCSV`

File: content/en/docs/Concepts/olm-architecture/operator-registry/building-a-catalog.md [hunk 3]
```diff
@@ -77,6 +77,106 @@ opm index add --bundles quay.io/my-container-registry-namespace/my-manifest-bund
...
+
+## SkipRange
+
+OLM also allows you to specify updates through version ranges in your CSV.
+
+```yaml
+metadata:
+   name: myoperator.v1.0.3
+   annotations:
+      olm.skiprange: <1.0.3
```
Comment (dinhxuanvu) [inline content/en/docs/Concepts/olm-architecture/operator-registry/building-a-catalog.md:143]: `skipRange`

File: content/en/docs/Concepts/olm-architecture/operator-registry/building-a-catalog.md [hunk 4]
```diff
@@ -77,6 +77,106 @@ opm index add --bundles quay.io/my-container-registry-namespace/my-manifest-bund
...
+```
+
+The version specifying the `skipRange` will be presented as a direct (one hop) update to
+any version from that package within that range. The versions in this range do not need to be in
+the index in order for bundle addition to be successful.
+
+**Warning:** Adding a bundle that only specifies skipRange to a given channel will wipe out all
+the previous content in that channel. This means directly installing past versions by editing
+the `startingCsv` field of the subscription is not possible when using skiprange only.
```
Comment (dinhxuanvu) [inline content/en/docs/Concepts/olm-architecture/operator-registry/building-a-catalog.md:152]: startingCSV

File: content/en/docs/Concepts/olm-architecture/operator-registry/building-a-catalog.md [hunk 5]
```diff
@@ -77,6 +77,106 @@ opm index add --bundles quay.io/my-container-registry-namespace/my-manifest-bund
...
+`myoperator.v1.0.3`.
+
+This is particularly useful if `myoperator.v1.0.1` and `myoperator.v1.0.2` are affected by a CVE
+or contain bugs.
+
+Skipped CSVs do not need to be present in an index prior to adding a given version.
+
+## SkipRange
+
+OLM also allows you to specify updates through version ranges in your CSV.
```
Comment (dinhxuanvu) [inline content/en/docs/Concepts/olm-architecture/operator-registry/building-a-catalog.md:137]: Should mention specifically the semver version range and link to semver version range spec page.

**Review by ankitathomas — APPROVED** [review]
This looks great, thanks for the great work @gallettilance! One small nit, but otherwise /lgtm

File: content/en/docs/Concepts/olm-architecture/operator-registry/building-a-catalog.md (hunk 3 above)
Comment (dinhxuanvu) [reply content/en/docs/Concepts/olm-architecture/operator-registry/building-a-catalog.md:143]: We usually avoid the example of unbounded range like this. You should use a bounded range example such as `>1.0.0 <1.1.0`.

File: content/en/docs/Concepts/olm-architecture/operator-registry/building-a-catalog.md [hunk 6]
```diff
@@ -77,6 +77,106 @@ opm index add --bundles quay.io/my-container-registry-namespace/my-manifest-bund
...
+
+The CSV name currently installed must be in the `replaces` field of a CSV in the new channel.
+
+### If using skips
+
+The CSV name currently installed must be in the `skips` field of a CSV in the new channel.
+
+### If using skipRange
+
+The version currently installed must be in the `skiprange` field of a CSV in the new channel.
```
Comment (dinhxuanvu) [inline content/en/docs/Concepts/olm-architecture/operator-registry/building-a-catalog.md:178]: `skipRange`

File: content/en/docs/Concepts/olm-architecture/operator-registry/building-a-catalog.md [hunk 7]
```diff
@@ -77,6 +77,106 @@ opm index add --bundles quay.io/my-container-registry-namespace/my-manifest-bund
...
+`myoperator.v1.0.3` without going through `myoperator.v1.0.1` or `myoperator.v1.0.2`. Installs
+that are already on `myoperator.v1.0.1` or `myoperator.v1.0.2` will still be able to update to
+`myoperator.v1.0.3`.
+
+This is particularly useful if `myoperator.v1.0.1` and `myoperator.v1.0.2` are affected by a CVE
+or contain bugs.
+
+Skipped CSVs do not need to be present in an index prior to adding a given version.
+
+## SkipRange
```
Comment (dinhxuanvu) [inline content/en/docs/Concepts/olm-architecture/operator-registry/building-a-catalog.md:135]: Should add a note/warning here is that CSV will need to specify `version` in order to use `skipRange`. That field is not required at the moment.

**Review by dinhxuanvu — CHANGES_REQUESTED** [review]
Just a few comments.

File: content/en/docs/Concepts/olm-architecture/operator-registry/building-a-catalog.md (hunk 2 above)
Comment (exdx) [inline content/en/docs/Concepts/olm-architecture/operator-registry/building-a-catalog.md:108]: should we also specify that the install plan approval setting on the sub should be set to `Manual` to ensure that it stays at that specific version and not auto-updates?

**Comment by exdx** [comment]
Also, I think this topic is so nuanced it deserves its own page, for example `How to specify an upgrade path for your operator` so we can easily link it to people when they have a question regarding skips/replaces. We could link to that page on the bottom of this page on building a catalog. WDYT?

**Review by ankitathomas — APPROVED** [review]
/lgtm

**Review by dinhxuanvu — APPROVED** [review]
/lgtm

**Review by anik120 — CHANGES_REQUESTED** [review]
@gallettilance thank you for writing up this document. 
I'm a bit confused about the document though. To begin with, I don't see it naturally fitting into the core tasks section of the site. The way the core task is set up is: "I am a brand new reader of this site and I have no idea of what olm is. I am going to follow these series of documents in the order they are in to learn what olm is about. So I go: 
Package my operator (i.e create my bundle) 
Validate my package   (i.e validate my bundle) 
Make my package part of a catalog (i.e load my bundle into an index) 
List operators available to install (i.e have the index loaded onto a cluster as a Catsrc) 
Install my operator (i.e create subscription) 
Uninstall my operator 
Uninstall olm"
This workflow might not come across right now coz a lot of those docs need to be updated to tell the latest story. 
With that in mind, maybe the best place for this information is in the document that talks about the [manifests](https://olm.operatorframework.io/docs/tasks/packaging-an-operator/), or at the very least, maybe we mention there olm allows for you to create a upgrade graph for your operator versions, explain what upgrade graph means (possible with an example diagram), and then point to a doc (possibly this one verbatim, or a little bit modified) that lives in the [concepts ](https://olm.operatorframework.io/docs/concepts/) section that details out ways to create that update graph? 

/hold

**Comment by gallettilance** [comment]
waiting on #89

**Review by dinhxuanvu — APPROVED** [review]
/lgtm

**Pull request merged**